src/requestManager.h

```diff
@@ -60,14 +60,14 @@ class CNodeRequestData
 public:
     int requestCount;
     int desirability;
-    CNode *node;
```

Comment:
A CNodeRef is effectively a hand-coded smart pointer to a CNode.  This change is creating a smart pointer to a smart pointer.  Removing CNodeRef (across the entire code base) would be a gigantic change, so I feel that removing the shared_ptr and using CNodeRef directly here would be cleaner and more efficient.

Reply:
done